<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en-US" xml:lang="en-US">
<head>
  <meta charset="utf-8" />
  <meta name="generator" content="pandoc" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
  <meta name="author" content="Casey S. Greene" />
  <meta name="author" content="Stephanie C. Hicks" />
  <meta name="author" content="Rob Patro" />
  <meta name="author" content="Elana J. Fertig" />
  <meta name="author" content="Michael I. Love" />
  <meta name="author" content="Jane Roe" />
  <meta name="dcterms.date" content="2018-11-02" />
  <meta name="keywords" content="dimensionality reduction, search, transformation, reference, cell types, single cell, Human Cell Atlas" />
  <title>Search for and transformation of human cells and cell types with latent space representations</title>
  <style type="text/css">
      code{white-space: pre-wrap;}
      span.smallcaps{font-variant: small-caps;}
      span.underline{text-decoration: underline;}
      div.column{display: inline-block; vertical-align: top; width: 50%;}
  </style>
  <link rel="stylesheet" href="github-pandoc.css" />
  <!--[if lt IE 9]>
    <script src="//cdnjs.cloudflare.com/ajax/libs/html5shiv/3.7.3/html5shiv-printshiv.min.js"></script>
  <![endif]-->
  <!-- Insert Analytics Script Below -->
  <script>
  </script>
  <!-- End Analytics Script -->
</head>
<body>
<header>
<h1 class="title">Search for and transformation of human cells and cell types with latent space representations</h1>
</header>
<p><small><em> This manuscript (<a href="https://greenelab.github.io/czi-seed-rfa/v/c49556f2dd2e5d5d7679f1753294bfb6ef3476f7/">permalink</a>) was automatically generated from <a href="https://github.com/greenelab/czi-seed-rfa/tree/c49556f2dd2e5d5d7679f1753294bfb6ef3476f7">greenelab/czi-seed-rfa@c49556f</a> on November 2, 2018. </em></small></p>
<h2 id="authors">Authors</h2>
<ul>
<li><p><strong>Casey S. Greene</strong><br> <img src="images/orcid.svg" alt="ORCID icon" width="13" height="13" /> <a href="https://orcid.org/0000-0001-8713-9213">0000-0001-8713-9213</a> · <img src="images/github.svg" alt="GitHub icon" width="13" height="13" /> <a href="https://github.com/cgreene">cgreene</a> · <img src="images/twitter.svg" alt="Twitter icon" width="13" height="13" /> <a href="https://twitter.com/greenescientist">greenescientist</a><br> <small> Department of Systems Pharmacology and Translational Therapeutics, Perelman School of Medicine, University of Pennsylvania </small></p></li>
<li><p><strong>Stephanie C. Hicks</strong><br> <img src="images/orcid.svg" alt="ORCID icon" width="13" height="13" /> <a href="https://orcid.org/0000-0002-7858-0231">0000-0002-7858-0231</a> · <img src="images/github.svg" alt="GitHub icon" width="13" height="13" /> <a href="https://github.com/stephaniehicks">stephaniehicks</a> · <img src="images/twitter.svg" alt="Twitter icon" width="13" height="13" /> <a href="https://twitter.com/stephaniechicks">stephaniechicks</a><br> <small> Department of Biostatistics, Johns Hopkins Bloomberg School of Public Health </small></p></li>
<li><p><strong>Rob Patro</strong><br> <img src="images/orcid.svg" alt="ORCID icon" width="13" height="13" /> <a href="https://orcid.org/0000-0001-8463-1675">0000-0001-8463-1675</a> · <img src="images/github.svg" alt="GitHub icon" width="13" height="13" /> <a href="https://github.com/rob-p">rob-p</a><br> <small> Department of Computer Science, Stony Brook University </small></p></li>
<li><p><strong>Elana J. Fertig</strong><br> <img src="images/orcid.svg" alt="ORCID icon" width="13" height="13" /> <a href="https://orcid.org/0000-0003-3204-342X">0000-0003-3204-342X</a> · <img src="images/github.svg" alt="GitHub icon" width="13" height="13" /> <a href="https://github.com/ejfertig">ejfertig</a> · <img src="images/twitter.svg" alt="Twitter icon" width="13" height="13" /> <a href="https://twitter.com/FertigLab">FertigLab</a><br> <small> Department of Oncology, Sidney Kimmel Comprehensive Cancer Center, School of Medicine, Johns Hopkins University; Department of Applied Mathematics and Statistics, Whiting School of Engineering, Johns Hopkins University </small></p></li>
<li><p><strong>Michael I. Love</strong><br> <img src="images/orcid.svg" alt="ORCID icon" width="13" height="13" /> <a href="https://orcid.org/0000-0001-8401-0545">0000-0001-8401-0545</a> · <img src="images/github.svg" alt="GitHub icon" width="13" height="13" /> <a href="https://github.com/mikelove">mikelove</a> · <img src="images/twitter.svg" alt="Twitter icon" width="13" height="13" /> <a href="https://twitter.com/mikelove">mikelove</a><br> <small> Department of Biostatistics, University of North Carolina at Chapel Hill; Department of Genetics, University of North Carolina at Chapel Hill </small></p></li>
<li><p><strong>Jane Roe</strong><br> <img src="images/orcid.svg" alt="ORCID icon" width="13" height="13" /> <a href="https://orcid.org/XXXX-XXXX-XXXX-XXXX">XXXX-XXXX-XXXX-XXXX</a> · <img src="images/github.svg" alt="GitHub icon" width="13" height="13" /> <a href="https://github.com/janeroe">janeroe</a><br> <small> Department of Something, University of Whatever; Department of Whatever, University of Something </small></p></li>
</ul>
<h2 id="abstract" class="page_break_before">Abstract</h2>
<p><strong>Instructions</strong>: Describe your collaborative project, highlighting key achievements of the project; limited to 250 words.</p>
<h2 id="five-key-references">Five Key References</h2>
<ul>
<li>Hicks refs: <span class="citation" data-cites="DJaucmAA">[<a href="#ref-DJaucmAA">1</a>]</span></li>
<li>projectR: <span class="citation" data-cites="cJPxOJMp">[<a href="#ref-cJPxOJMp">2</a>]</span></li>
<li>Alevin: <span class="citation" data-cites="FPpU83vH">[<a href="#ref-FPpU83vH">3</a>]</span></li>
</ul>
<h2 id="project-team">Project Team</h2>
<h3 id="pi-information">PI information</h3>
<ol type="1">
<li><p>Loyal Goff (Submitter)</p></li>
<li><p>Stephanie Hicks</p>
<ul>
<li>Title: Assistant Professor</li>
<li>Degrees: PhD</li>
<li>Type of organization: Academic</li>
<li>Tax ID: 52-0595110 (JHU)</li>
<li>Email: shicks19@jhu.edu</li>
</ul></li>
<li><p>Elana Fertig</p>
<ul>
<li>Title: Associate Professor</li>
<li>Degrees: PhD</li>
<li>Type of organization: Academic</li>
<li>Tax ID: 52-0595110 (JHU)</li>
<li>Email: ejfertig@jhmi.edu</li>
</ul></li>
<li><p>Casey Greene</p></li>
<li><p>Tom Hampton</p></li>
<li><p>Michael Love</p>
<ul>
<li>Title: Assistant Professor</li>
<li>Degrees: Dr. rer. nat.</li>
<li>Type of organization: Academic</li>
<li>Tax ID: 56-6001393 (UNC)</li>
<li>Email: milove@email.unc.edu</li>
</ul></li>
<li>Rob Patro
<ul>
<li>Title: Assistant Professor</li>
<li>Degrees: PhD</li>
<li>Type of Organization: Academic</li>
<li>Tax ID: 16-1514621 (Stony Brook)</li>
<li>Email: rob.patro@cs.stonybrook.edu</li>
</ul></li>
</ol>
<h3 id="description-750-words-total">Description (750 words TOTAL)</h3>
<ol type="1">
<li>Loyal Goff</li>
<li>Stephanie C. Hicks is an Assistant Professor of Biostatistics at the Johns Hopkins Bloomberg School of Public Health. She is an expert in statistical methodology with a strong track record in processing and analyzing single-cell genomics data, including extensive experience developing fast, memory-efficient R/Bioconductor software to remove systematic and technical biases from scRNA-seq data <span class="citation" data-cites="DJaucmAA">[<a href="#ref-DJaucmAA">1</a>]</span>. Dr. Hicks will work together with Co-PIs to implement fast search algorithms in latent spaces (Aim 1) and to implement the methods developed into fast, scalable, and memory-efficient R/Bioconductor software packages (Aim 3).</li>
<li>Elana Fertig is an Associate Professor of Oncology and Applied Mathematics and Statistics at Johns Hopkins University. She developed of the Bayesian non-negative matrix factorization algorithm CoGAPS <span class="citation" data-cites="1DrhKLdVp">[<a href="#ref-1DrhKLdVp">4</a>]</span> for latent space analysis. In collaboration with co-PI Goff, she adapted this tool to scRNA-seq data and developed a new transfer learning framework to relate the low-dimensional features in scRNA-seq data across data modalities, biological conditions, and organisms <span class="citation" data-cites="cJPxOJMp">[<a href="#ref-cJPxOJMp">2</a>]</span>. Dr. Fertig will work with the co-PIs to incorporate the error models from Aim 1 into the latent space representations, dimensionality estimation, and biological assessment metrics in Aim 2. She is developing standardized language for latent space representation in collaboration with co-PIs Goff and Greene <span class="citation" data-cites="Sn52lYwa">[<a href="#ref-Sn52lYwa">5</a>]</span> that will provide a strong foundation for standardization of these approaches across different unsupervised learning tools.</li>
<li>Casey Greene</li>
<li>Tom Hampton</li>
<li>Michael Love is an Assistant Professor of Biostatistics and Genetics at the University of North Carolina at Chapel Hill. He is a leading developer of statistical software for RNA-seq analysis in the Bioconductor Project, maintaining the widely used DESeq2 <span class="citation" data-cites="w9AOzBMw">[<a href="#ref-w9AOzBMw">6</a>]</span> and tximport <span class="citation" data-cites="9CN5KEFo">[<a href="#ref-9CN5KEFo">7</a>]</span> packages. He is a close collaborator with Dr. Rob Patro on bias-aware estimation of transcript abundance from RNA-seq and estimation of uncertainty during transcript quantification <span class="citation" data-cites="vrqQcFyx">[<a href="#ref-vrqQcFyx">8</a>]</span>. Dr. Love will work with co-PIs to disseminate versioned reference cell type catalogs through widely used frameworks for genomic data analysis including R/Bioconductor and Python.</li>
<li>Rob Patro</li>
</ol>
<h2 id="proposal-body-2000-words">Proposal Body (2000 words)</h2>
<p>Latent space determination relies on computational techniques that learn low dimensional structure from high dimensional data. A wide variety of computational methods for dimensionality reduction have been applied to single cell data. These techniques fall largely into two classes: (1) linear techniques based upon matrix factorization and (2) non-linear techniques based upon manifold learning. Each specific technique selected for analysis will reveal distinct structures in the data. Much of the current research emphasizes demonstration of optimality of a single method. In contrast to this prevailing view, we posit that each method reveals a unique set of features associated with a set of <em>a priori</em> unknown biological processes. Together, the distribution of the ensemble of low dimensional representations from these methods will uncover the multiple, concurrent biological processes from high dimensional datasets. Previously, we have developed a common language for interpretation of matrix factorization methods to facilitate a unification of latent space methods <span class="citation" data-cites="Sn52lYwa">[<a href="#ref-Sn52lYwa">5</a>]</span>. The latent space team in the previous round of funding (containing PIs Fertig, Goff, Greene, and Patro) is also developing standardized output formats for these distinct methods. Together, this previous work will facilitate practical unification and biological assessment of latent spaces in this seed network.</p>
<p>Numerous factors impact the biological accuracy of latent spaces learned from dimension reduction techniques. Two notable examples include the preprocessing of the input data (Aim 1) and dimensionality of the low dimensional representation (Aim 2). The impact of each of these components on inference must be quantified and isolated before latent methods can be compared or unified. To that end, we select a single dimension reduction technique as the foundation for the technologies developed in our seed network. We use the Bayesian, non-negative matrix factorization method scCoGAPS <span class="citation" data-cites="6i1NIkNx">[<a href="#ref-6i1NIkNx">9</a>]</span> (PI Fertig). Previous work has demonstrated the robustness of this approach in inferring biologically relevant low dimensional representations of perturbation <span class="citation" data-cites="wkhRfjyx">[<a href="#ref-wkhRfjyx">10</a>]</span> and time course <span class="citation" data-cites="uInnOMwX">[<a href="#ref-uInnOMwX">12</a>]</span> from bulk datasets, leading to the winning solution in the HPN DREAM8 challenge <span class="citation" data-cites="qpg6x7P4">[<a href="#ref-qpg6x7P4">14</a>]</span>. Moreover, recent extension of this method to single cell data in the developing mouse retina distinguished simultaneously cellular identity, dynamic trajectories, and cell state within a single analysis <span class="citation" data-cites="6i1NIkNx">[<a href="#ref-6i1NIkNx">9</a>]</span>. Notably for the work proposed in this seed network, this method includes an uncertainty estimate that can be readily modified to account for measurement-specific technical variation <span class="citation" data-cites="5Cj8i4Xu">[<a href="#ref-5Cj8i4Xu">15</a>]</span>. Moreover, this same study demonstrated that low dimensional factorizations of tumor populations separate tumor samples relative to normals whereas higher dimensionalities separate tumor subtypes. Thus, CoGAPS has been shown to yield different factorizations reflective of biological hierarchies at different dimensionalities, rather than having a single dimensionality with a “correct” factorization. While technology development will focus on scCoGAPS, the tools developed as part of the seed network will be generalizable and compared to additional linear and non-linear methods developed in the literature <strong>ADD RELEVANT CITATIONS</strong> and by members of the broader CZI consortium. This single-method focus with subsequent expansion to additional methods will also facilitate educational initiatives to advance broader understanding of the interpretation of latent space solutions from high dimensional data (Aim 3).</p>
<p><em>About fast and improved quant</em> Existing approaches for quantification from scRNA-seq data using tagged-end end protocols (e.g. 10x Chromium, drop-Seq, inDrop, etc.) have no mechanism for accounting for reads mapping between multiple genes in the resulting quantification estimates, and therefore such reads are typically removed <strong>CORRECT?</strong>. This reduces quantification accuracy, and leads to systematic biases in gene expression estimates that correlate problematically with the size of gene families and gene function <span class="citation" data-cites="FPpU83vH">[<a href="#ref-FPpU83vH">3</a>]</span> . We have recently developed a novel approach for quantification from tagged-end data that accounts for reads mapping to multiple genomic loci (~15-25% of the reads in a typical experiment) in a principled and consistent way. We propose to expand on this work in a number of ways, building these capabilities into a production quality tool for the processing of scRNA-seq data. (1) Exploration of alternative models for UMI resolution, including a maximum likelihood (as opposed to maximum parsimony) model, (2) Development of new approaches for quality-control and filtering using features from the the UMI-resolution graph, (3) Creation of a compressed and indexible data structure for the UMI-resolution graph to allow direct access, query, and, eventually, search.</p>
<p>The technologies to improve quantification will have a critical impact on the outcomes of latent spaces. However, there are currently no standardized, quantitative metrics to determine relative uncovering of biology from low dimensional representations. We have developed new transfer learning methods to quantify the extent to which latent space representations from one set of training data are represented in another <span class="citation" data-cites="cJPxOJMp">[<a href="#ref-cJPxOJMp">2</a>]</span>. These tools provide a strong foundation to enable biological quantification of latent space representations by quantifying the extent to which those spaces transfer across datasets of related biological contexts.</p>
<h3 id="aim-1">Aim 1</h3>
<p><em>Rationale</em>: need to add</p>
<p><em>Fast search algorithms</em> (in low dimensions, quantifying differences between case and reference maps, twist: shared latent spaces / k-mers):</p>
<p><em>Finding differences between maps</em>: The fast search algorithms will then be used quantify differences between a reference transcriptome map (the Human Cell Atlas) and non-reference transcriptome maps from other samples of interest, similar to the idea of using a reference genome to identify genomic differences in between a reference and non-reference genome. Globally quantifying differences between transcriptome maps is important because it allows for quantification of differences at the population or individual level between, for example, ten transcriptome maps from individuals with a particular phenotype to be compared to the Human Cell Atlas reference map (or other control transcriptome maps if available). Our metric to quantify differences will depend on the distributions of cell expression within and between individuals, which Dr. Hicks has extensive experience with <span class="citation" data-cites="13owodqhx">[<a href="#ref-13owodqhx">17</a>]</span>. We will leverage not only the cell-to-cell correlation structure within one transcriptome map (or human individual), but also the correlation structure across transcriptome maps (or multiple human individuals), which will share common latent spaces across individuals for a particular phenotype. Our initial approach will be to use linear mixed models to account for the correlation structure within and between transcriptome maps. The statistical method will be fast, memory-efficient and will scale to billions of cells because we work in the latent space with a significantly reduced number of dimensions (instead of billions, just hundreds or thousands).</p>
<p><em>New models for UMI deduplication accounting for transcript-level information</em>: (Rob)</p>
<p><em>Parsimony &amp; likelihood based, integrated with gene-level uncertainty</em>:</p>
<pre><code>* Everything FAST! API for search against HCA reference? (Rob)
* k-mer / quantified latent spaces (Casey / Rob)</code></pre>
<h3 id="aim-2">Aim 2</h3>
<p><em>Rationale:</em> Biological systems are comprised of common cell types with overlapping molecular phenotypes. The function of these systems are determined by the interactions between these complex components, rather than a single gene or cell. This suggests that fundamental biological mechanisms may broadly contribute to an observed state, with context-specific modifiers conferring selective suceptiability to disease. Latent space techniques are poised to reveal these fundamental mechanisms in the broad survey of single cell data across model systems and cellular contexts in the Human Cell Atlas. We hypothesize that the features learned from these techniques will define constitutive basis vectors that reflect discrete biological processes or features. Thus, these basis vectors will be shared across different biological systems, with context-specific perturbations indicating pathogenic differences in disease. <em>We propose a central suite of statistics for assessment and interpretation of latent space tools to define the basis vectors of biological systems.</em></p>
<p><em>Quantifying latent space estimation with transfer learning:</em> A critical challenge to latent space methods is the quantification of methods performance. Numerous computational metrics have been developed to assess convergence of the low dimensional estimation. However, these metrics do not quantify whether the features in a low dimensional representation of scRNA-seq data represent biological processes in the measured system. The performance of these methods can be quantified directly in datasets for which cell types and states are known (e.g., perturbation experiments, controlled admixture experiments, etc). However, these annotations are lacking in most biological datasets limiting any such quantification. Transfer learning methods have been developed in machine learning to relate features learned in a source dataset to those in a new, target dataset in order to transfer annotations from one context to another. In this project, we will adapt these methods to quantify the performance of latent space methods by the extent to which learned low dimensional features from a source dataset transfer to a target dataset in a related biological context. We will benchmark the performance of the resulting metric on simulated datasets, cross-validation in scRNA-seq datasets with known cell types and states, and cross-study validation of systems in related biological contexts with known cell types and states. Gene set enrichment methods will also be used to explore the relevant biological processes described by individual basis vectors, and related bases will be identified through clustering and exploratory approaches in these benchmark datasets. Our transfer learning based metric will be piloted on low dimensional representations learned with scCoGAPS and then applied to a broader suite of latent space tools. We will release software for this transfer learning quantification of latent space representations in R and Python using standard latent space file formats developed by our team in the first year of HCA funding.</p>
<p><em>Dimensionality estimation:</em> Dimensionality reduction methods are sensitive to the number of low features learned in each dataset. Many computational techniques optimize dimensionality by creating a cost function which penalizes models with higher number of features. Similar to the quantification metrics, these penalty terms do not reflect the extent to which features learned at a given dimensionality reflect biology. Moreover, many systems may have more than one biologically accurate low dimensional representation. Such multiple truths in data would be particular prominent in systems that can be subdivided into hierarchical classifications. For example, in the case of cancer we observed that a low dimensional representation of bulk data learned from CoGAPS distinguished cancers from normals whereas a higher dimension distinguished tumor subtypes <span class="citation" data-cites="5Cj8i4Xu">[<a href="#ref-5Cj8i4Xu">15</a>]</span>. Both of these low dimensional representations are equally valid, and each reflects different biological features in the data. To find these multiple truths, we will develop a parallel framework to run scCoGAPS for multiple dimensionalities and quantify performance with our transfer-learning based metric on random subsets of the data. The dimensions with greatest cross-validated feature robustness will be retained as the optimal dimensionalities for each dataset. We will develop software to enable this cross-validation dimensionality estimation across multiple latent space methods. We note that this same software will provide a robust tool to define ensembles of low dimensional representations that reflect underlying biology learned across multiple latent space methods. <strong>Rob: I’m not sure if you want to fill in some of your ideas re persistent homology instead. Very open to that idea and think it may be a nice, more efficient methodology than what’s proposed here.</strong></p>
<p><em>Search tool for latent spaces and reference cell types:</em> <strong>Loyal, Casey – what are the datasets that will be used for this – I would think all healthy cells in a single system to enable quantification of context-specific in the next part of this aim.</strong> Comprehensive identification of basis vectors across conditions is an area of active research for our group in the previous funding period. We will use scCoGAPS and other tools developed within our collaborative network to establish a compendium of basis vectors across our single cell catalog. Ensembles of the low dimensional features that represent robust biological features across methods using methods described above will be preserved as the ‘biological basis’ of the Human Cell Atlas. The weights of these bases will be correlated across all available metadata attributes for each cell to identify basis vectors that are associated with specific cellular contexts, disease states, technical parameters, or other phenotypic features. A reference catalog of gene weights for specific cell types will be defined by the set of basis vectors associated with cellular identity in datasets with known ground truth. We will adapt the software we developed for transfer learning of features from bulk data recount <span class="citation" data-cites="1GtRgPRxn">[<a href="#ref-1GtRgPRxn">16</a>]</span> to facilitate querying of signatures in new user-defined datasets (delivery of which is described in the next aim). As datasets accumulate and methods are refined, the biological basis and reference catalog of gene weights will evolve over time. To enable reproducible research leveraging HCA, we will implement a content-based versioning system, which identifies versions of the reference cell type catalog by the gene weights and transcript nucleotide sequences using a hash function. Such a hash-based versioning and provenance identification and detection framework has proven successful in the bulk RNA-seq context to support reproducible computational analyses <span class="citation" data-cites="1FQ0kp4Dj">[<a href="#ref-1FQ0kp4Dj">18</a>]</span>.</p>
<p><em>Differentiating context-specific latent spaces from latent spaces that are universal across biological contexts:</em> The search tool to define reference cell types based upon latent spaces was defined for healthy tissues from XXX (some control). Deviations of common cell types or states from the healthy baseline in other populations will indicate context-specific alterations, which may be associated with disease. To identify potentially pathogenic responses in target datasets, we will implement a random forest classifier into our transfer learning method to segregate cells based on their usage of disease-associated basis vectors after projection. In other cases, disease may arise from changes in variation reflective of inter-cellular heterogeneity. Therefore, we will also develop methods to quantify variation from latent space vectors. Both methods will be incorporated in our latent space search tool. <strong>Loyal: I’m not sure if this is what you had in mind. It may also be that these are reflected in the hierarchy of dimensionality – may want to incorporate here.</strong></p>
<h3 id="aim-3">Aim 3</h3>
<p><em>Rationale:</em> Low dimensional representations provide a powerful means to analyze scRNA-seq and HCA data to make tasks faster, perform more biologically grounded analyses, and provide interpretable summaries of complex high-dimensional data. However, using these capabilities to the fullest extent requires an investment in robustness and evaluation, integration with widely used toolkits, and in a scalable education effort that can reach students from the undergraduate level and beyond. <em>We propose to enhance software usability and develop open instructional materials that we will use to deliver short-course training that includes the topics of single cell profiling, machine learning methods, low-dimensional representations, reference cell type catalogs, and tools developed by our group in response to this RFA.</em></p>
<h4 id="aim-3.1-training-for-next-generation-of-single-cell-data-scientists-tom">Aim 3.1: Training for next-generation of single cell data scientists (Tom)</h4>
<p>Training / teaching (scRNAseq, low-dimensional representations, reference cell type catalog, RFA-developed tools)</p>
<p>Although the HCA data set will greatly increase the rate of biological discovery across many biomedical fields, a background in bioinformatics will be required to these data effectively. Our mission is to make this obstacle easy for most biologists and physician scientists to overcome, and to enable scientists who already possess a background in bioinformatics to use software created in this project effectively. We will address these goals by offering short courses and creating distributed communities.</p>
<ol type="1">
<li><p>Short Courses. Short courses (3-5 days) will introduce all features of bioinformatics required to access HCA and use the tools developed in this application successfully, such as:</p>
<ul>
<li>Visualization and Exploration of High Dimensional Data</li>
<li>R Statistical Programming Environment</li>
<li>UNIX</li>
<li>Statistical Approaches for High Dimension Biomedical Data</li>
<li>Gene Set and Pathway Analysis</li>
<li>Low dimensional representations of high dimensional data</li>
<li>R and UNIX Tools Specific to HCA Data Analysis</li>
<li>scRNAseq tools</li>
<li>RFA-developed tools</li>
</ul></li>
<li><p>Course Project. The course project will provide a hands-on experience of retrieving data from HCA, performing quality control, exploratory data analysis, identification of differentially abundant clusters, identification of martker genes associated with clusters, differential gene expression, and interpretation of results in the context of other data such as Gene Ontology and KEGG pathways. Groups of participants share their findings in a final presentation.</p></li>
<li><p>Distributed Communities. Where short courses jump start participants’ ability to perform basic tasks and learn the tools relevant to scRNA-seq, distributed communities provide the opportunity of mastery of these techniques, which includes the ability to teach others. Our support of these communities will include:</p>
<ul>
<li>Workshops and Seminars For Group Leaders</li>
<li>Access to Community-Developed Curricula and Documentation</li>
<li>Scholarships for Short Courses</li>
</ul></li>
</ol>
<h4 id="aim-3.2-software-engineering-and-integration-into-bioconductor-stephanie-and-casey">Aim 3.2: Software Engineering and Integration into Bioconductor (Stephanie and Casey)</h4>
<p>We will implement the proposed methods from Aims 1 and 2 into robust software, which will be integrated into the R/Bioconductor and Python frameworks. The software will be fast, scalable, and memory-efficient because will leverage the computational tools previously developed by Bioconductor for single-cell data access to the HCA, data representation (<code>SingleCellExperiment</code>, <code>beachmat</code>, <code>DelayedArray</code>, <code>HDF5Array</code> and <code>rhdf5</code>) and data assessment and ameliorization of data quality (<code>scater</code>, <code>scran</code>, <code>DropletUtils</code>).</p>
<h4 id="aim-3.3-data-integration-into-bioconductor-stephanie-and-mike">Aim 3.3: Data Integration into Bioconductor (Stephanie and Mike)</h4>
<p>We will integrate the catalogs of reference cell types (Aim 2) into the R/Bioconductor and Python frameworks. The use of consortia data summaries as annotation and interpretive scaffold for outside datasets has proven widely successful for the ENCODE, Roadmap Epigenome Mapping, and GTEx projects. We will use HCA data to define similar summaries as annotation and scaffolding for local genomic datasets. Additional information about biological variability that can be transferred from individual or ensemble latent spaces, can also be used landmarks to ordinate differences among populations of cells observed in a local datasets, whether bulk or single-cell experiments. We will package and version reference cell types, including measures of technical and biological variability transferred from the latent spaces defined in earlier aims, and deliver these as structured data objects in Bioconductor and Python. We will leverage our expertise in working with Bioconductor both as core package developers and power users to enable on-the-fly downloading of reference cell types and variability via the <em>AnnotationHub</em> framework, which includes rich specification of metadata including the provenance and versioning of the catalog. Python workflows leveraging existing frameworks for genomic data analysis (Biopython, BioRanges, and bx-pythong) will also be supported via cross-platform packages such as Feather that facilitate transfer of data objects between languages. We will develop <em>F1000Research</em> workflows for R and python demonstrating how HCA-defined reference cell types and tools developed in this RFA can be used within a typical genomic data analysis.</p>
<h2 id="references" class="page_break_before">References</h2>
<!-- Explicitly insert bibliography here -->
<div id="refs">
<div id="ref-DJaucmAA">
<p>1. <strong>Missing data and technical variability in single-cell RNA-sequencing experiments</strong><br />
Stephanie C Hicks, F William Townes, Mingxiang Teng, Rafael A Irizarry<br />
<br />
<em>Biostatistics</em> (2017-11-06) <a href="https://doi.org/gfb8g4" class="uri">https://doi.org/gfb8g4</a><br />
DOI: <a href="https://doi.org/10.1093/biostatistics/kxx053">10.1093/biostatistics/kxx053</a> · PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/29121214">29121214</a></p>
</div>
<div id="ref-cJPxOJMp">
<p>2. <strong>Decomposing cell identity for transfer learning across cellular measurements, platforms, tissues, and species.</strong><br />
Genevieve L Stein-O’Brien, Brian S. Clark, Thomas Sherman, Christina Zibetti, Qiwen Hu, Rachel Sealfon, Sheng Liu, Jiang Qian, Carlo Colantuoni, Seth Blackshaw, … Elana J. Fertig<br />
<br />
<em>Cold Spring Harbor Laboratory</em> (2018-08-20) <a href="https://doi.org/gd2xpn" class="uri">https://doi.org/gd2xpn</a><br />
DOI: <a href="https://doi.org/10.1101/395004">10.1101/395004</a></p>
</div>
<div id="ref-FPpU83vH">
<p>3. <strong>Alevin efficiently estimates accurate gene abundances from dscRNA-seq data</strong><br />
Avi Srivastava, Laraib Malik, Tom Sean Smith, Ian Sudbery, Rob Patro<br />
<br />
<em>Cold Spring Harbor Laboratory</em> (2018-06-01) <a href="https://doi.org/gffk42" class="uri">https://doi.org/gffk42</a><br />
DOI: <a href="https://doi.org/10.1101/335000">10.1101/335000</a></p>
</div>
<div id="ref-1DrhKLdVp">
<p>4. <strong>CoGAPS: an R/C++ package to identify patterns and biological process activity in transcriptomic data</strong><br />
Elana J. Fertig, Jie Ding, Alexander V. Favorov, Giovanni Parmigiani, Michael F. Ochs<br />
<br />
<em>Bioinformatics</em> (2010-09-01) <a href="https://doi.org/cwqsv4" class="uri">https://doi.org/cwqsv4</a><br />
DOI: <a href="https://doi.org/10.1093/bioinformatics/btq503">10.1093/bioinformatics/btq503</a> · PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/20810601">20810601</a> · PMCID: <a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3025742">PMC3025742</a></p>
</div>
<div id="ref-Sn52lYwa">
<p>5. <strong>Enter the Matrix: Factorization Uncovers Knowledge from Omics</strong><br />
Genevieve L. Stein-O’Brien, Raman Arora, Aedin C. Culhane, Alexander V. Favorov, Lana X. Garmire, Casey S. Greene, Loyal A. Goff, Yifeng Li, Aloune Ngom, Michael F. Ochs, … Elana J. Fertig<br />
<br />
<em>Trends in Genetics</em> (2018-10) <a href="https://doi.org/gd93tk" class="uri">https://doi.org/gd93tk</a><br />
DOI: <a href="https://doi.org/10.1016/j.tig.2018.07.003">10.1016/j.tig.2018.07.003</a> · PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/30143323">30143323</a></p>
</div>
<div id="ref-w9AOzBMw">
<p>6. <strong>Moderated estimation of fold change and dispersion for RNA-seq data with DESeq2</strong><br />
Michael I Love, Wolfgang Huber, Simon Anders<br />
<br />
<em>Genome Biology</em> (2014-12) <a href="https://doi.org/gd3zvn" class="uri">https://doi.org/gd3zvn</a><br />
DOI: <a href="https://doi.org/10.1186/s13059-014-0550-8">10.1186/s13059-014-0550-8</a> · PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/25516281">25516281</a> · PMCID: <a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4302049">PMC4302049</a></p>
</div>
<div id="ref-9CN5KEFo">
<p>7. <strong>Differential analyses for RNA-seq: transcript-level estimates improve gene-level inferences</strong><br />
Charlotte Soneson, Michael I. Love, Mark D. Robinson<br />
<br />
<em>F1000Research</em> (2015-12-30) <a href="https://doi.org/gdtgw8" class="uri">https://doi.org/gdtgw8</a><br />
DOI: <a href="https://doi.org/10.12688/f1000research.7563.1">10.12688/f1000research.7563.1</a> · PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/26925227">26925227</a> · PMCID: <a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4712774">PMC4712774</a></p>
</div>
<div id="ref-vrqQcFyx">
<p>8. <strong>Salmon provides fast and bias-aware quantification of transcript expression</strong><br />
Rob Patro, Geet Duggal, Michael I Love, Rafael A Irizarry, Carl Kingsford<br />
<br />
<em>Nature Methods</em> (2017-03-06) <a href="https://doi.org/gcw9f5" class="uri">https://doi.org/gcw9f5</a><br />
DOI: <a href="https://doi.org/10.1038/nmeth.4197">10.1038/nmeth.4197</a> · PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/28263959">28263959</a> · PMCID: <a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC5600148">PMC5600148</a></p>
</div>
<div id="ref-6i1NIkNx">
<p>9. <strong>Comprehensive analysis of retinal development at single cell resolution identifies NFI factors as essential for mitotic exit and specification of late-born cells</strong><br />
Brian Clark, Genevieve Stein-O’Brien, Fion Shiau, Gabrielle Cannon, Emily Davis, Thomas Sherman, Fatemeh Rajaii, Rebecca James-Esposito, Richard Gronostajski, Elana Fertig, … Seth Blackshaw<br />
<br />
<em>Cold Spring Harbor Laboratory</em> (2018-07-27) <a href="https://doi.org/gdwrzh" class="uri">https://doi.org/gdwrzh</a><br />
DOI: <a href="https://doi.org/10.1101/378950">10.1101/378950</a></p>
</div>
<div id="ref-wkhRfjyx">
<p>10. <strong>Gene expression signatures modulated by epidermal growth factor receptor activation and their relationship to cetuximab resistance in head and neck squamous cell carcinoma</strong><br />
Elana J Fertig, Qing Ren, Haixia Cheng, Hiromitsu Hatakeyama, Adam P Dicker, Ulrich Rodeck, Michael Considine, Michael F Ochs, Christine H Chung<br />
<br />
<em>BMC Genomics</em> (2012) <a href="https://doi.org/gb3fgp" class="uri">https://doi.org/gb3fgp</a><br />
DOI: <a href="https://doi.org/10.1186/1471-2164-13-160">10.1186/1471-2164-13-160</a> · PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/22549044">22549044</a> · PMCID: <a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3460736">PMC3460736</a></p>
</div>
<div id="ref-1GkdWBzqU">
<p>11. <strong>CoGAPS matrix factorization algorithm identifies transcriptional changes in AP-2alpha target genes in feedback from therapeutic inhibition of the EGFR network</strong><br />
Elana J. Fertig, Hiroyuki Ozawa, Manjusha Thakar, Jason D. Howard, Luciane T. Kagohara, Gabriel Krigsfeld, Ruchira S. Ranaweera, Robert M. Hughes, Jimena Perez, Siân Jones, … Christine H. Chung<br />
<br />
<em>Oncotarget</em> (2016-09-16) <a href="https://doi.org/f9k8d8" class="uri">https://doi.org/f9k8d8</a><br />
DOI: <a href="https://doi.org/10.18632/oncotarget.12075">10.18632/oncotarget.12075</a> · PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/27650546">27650546</a> · PMCID: <a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC5342018">PMC5342018</a></p>
</div>
<div id="ref-uInnOMwX">
<p>12. <strong>Pattern Identification in Time-Course Gene Expression Data with the CoGAPS Matrix Factorization</strong><br />
Elana J. Fertig, Genevieve Stein-O’Brien, Andrew Jaffe, Carlo Colantuoni<br />
<br />
<em>Gene Function Analysis</em> (2013-10-24) <a href="https://doi.org/f5j7xj" class="uri">https://doi.org/f5j7xj</a><br />
DOI: <a href="https://doi.org/10.1007/978-1-62703-721-1_6">10.1007/978-1-62703-721-1_6</a> · PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/24233779">24233779</a></p>
</div>
<div id="ref-1DZRsfkoC">
<p>13. <strong>Integrated time course omics analysis distinguishes immediate therapeutic response from acquired resistance</strong><br />
Genevieve Stein-O’Brien, Luciane T. Kagohara, Sijia Li, Manjusha Thakar, Ruchira Ranaweera, Hiroyuki Ozawa, Haixia Cheng, Michael Considine, Sandra Schmitz, Alexander V. Favorov, … Elana J. Fertig<br />
<br />
<em>Genome Medicine</em> (2018-05-23) <a href="https://doi.org/gfc4dq" class="uri">https://doi.org/gfc4dq</a><br />
DOI: <a href="https://doi.org/10.1186/s13073-018-0545-2">10.1186/s13073-018-0545-2</a> · PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/29792227">29792227</a> · PMCID: <a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC5966898">PMC5966898</a></p>
</div>
<div id="ref-qpg6x7P4">
<p>14. <strong>Inferring causal molecular networks: empirical assessment through a community-based effort</strong><br />
Steven M HillLaura M Heiser, Thomas Cokelaer, Michael Unger, Nicole K Nesser, Daniel E Carlin, Yang Zhang, Artem Sokolov, Evan O Paull, … Sach Mukherjee<br />
<br />
<em>Nature Methods</em> (2016-02-22) <a href="https://doi.org/f3t7t4" class="uri">https://doi.org/f3t7t4</a><br />
DOI: <a href="https://doi.org/10.1038/nmeth.3773">10.1038/nmeth.3773</a> · PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/26901648">26901648</a> · PMCID: <a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4854847">PMC4854847</a></p>
</div>
<div id="ref-5Cj8i4Xu">
<p>15. <strong>Preferential Activation of the Hedgehog Pathway by Epigenetic Modulations in HPV Negative HNSCC Identified with Meta-Pathway Analysis</strong><br />
Elana J. Fertig, Ana Markovic, Ludmila V. Danilova, Daria A. Gaykalova, Leslie Cope, Christine H. Chung, Michael F. Ochs, Joseph A. Califano<br />
<br />
<em>PLoS ONE</em> (2013-11-04) <a href="https://doi.org/gcpgc6" class="uri">https://doi.org/gcpgc6</a><br />
DOI: <a href="https://doi.org/10.1371/journal.pone.0078127">10.1371/journal.pone.0078127</a> · PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/24223768">24223768</a> · PMCID: <a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3817178">PMC3817178</a></p>
</div>
<div id="ref-1GtRgPRxn">
<p>16. <strong>MultiPLIER: a transfer learning framework reveals systemic features of rare autoimmune disease</strong><br />
Jaclyn N Taroni, Peter C Grayson, Qiwen Hu, Sean Eddy, Matthias Kretzler, Peter A Merkel, Casey S Greene<br />
<br />
<em>Cold Spring Harbor Laboratory</em> (2018-08-20) <a href="https://doi.org/gfc9bb" class="uri">https://doi.org/gfc9bb</a><br />
DOI: <a href="https://doi.org/10.1101/395947">10.1101/395947</a></p>
</div>
<div id="ref-13owodqhx">
<p>17. <strong>quantro: a data-driven approach to guide the choice of an appropriate normalization method.</strong><br />
Stephanie C Hicks, Rafael A Irizarry<br />
<br />
<em>Genome biology</em> (2015-06-04) <a href="https://www.ncbi.nlm.nih.gov/pubmed/26040460" class="uri">https://www.ncbi.nlm.nih.gov/pubmed/26040460</a><br />
DOI: <a href="https://doi.org/10.1186/s13059-015-0679-0">10.1186/s13059-015-0679-0</a> · PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/26040460">26040460</a> · PMCID: <a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4495646">PMC4495646</a></p>
</div>
<div id="ref-1FQ0kp4Dj">
<p>18. <strong>tximeta</strong><br />
Rob Patro Michael Love<br />
<br />
<em>Bioconductor</em> (2018) <a href="https://doi.org/gfddxw" class="uri">https://doi.org/gfddxw</a><br />
DOI: <a href="https://doi.org/10.18129/b9.bioc.tximeta">10.18129/b9.bioc.tximeta</a></p>
</div>
</div>
<script>
// AnchorJS minified version below.
// Source https://github.com/bryanbraun/anchorjs/blob/064abdd0987f305933ec4982af6d0c1cf2fd0814/anchor.js

/**
 * AnchorJS - v4.0.0 - 2017-06-02
 * https://github.com/bryanbraun/anchorjs
 * Copyright (c) 2017 Bryan Braun; Licensed MIT
 */
!function(A,e){"use strict";"function"==typeof define&&define.amd?define([],e):"object"==typeof module&&module.exports?module.exports=e():(A.AnchorJS=e(),A.anchors=new A.AnchorJS)}(this,function(){"use strict";function A(A){function e(A){A.icon=A.hasOwnProperty("icon")?A.icon:"",A.visible=A.hasOwnProperty("visible")?A.visible:"hover",A.placement=A.hasOwnProperty("placement")?A.placement:"right",A.class=A.hasOwnProperty("class")?A.class:"",A.truncate=A.hasOwnProperty("truncate")?Math.floor(A.truncate):64}function t(A){var e;if("string"==typeof A||A instanceof String)e=[].slice.call(document.querySelectorAll(A));else{if(!(Array.isArray(A)||A instanceof NodeList))throw new Error("The selector provided to AnchorJS was invalid.");e=[].slice.call(A)}return e}function n(){if(null===document.head.querySelector("style.anchorjs")){var A,e=document.createElement("style");e.className="anchorjs",e.appendChild(document.createTextNode("")),void 0===(A=document.head.querySelector('[rel="stylesheet"], style'))?document.head.appendChild(e):document.head.insertBefore(e,A),e.sheet.insertRule(" .anchorjs-link {   opacity: 0;   text-decoration: none;   -webkit-font-smoothing: antialiased;   -moz-osx-font-smoothing: grayscale; }",e.sheet.cssRules.length),e.sheet.insertRule(" *:hover > .anchorjs-link, .anchorjs-link:focus  {   opacity: 1; }",e.sheet.cssRules.length),e.sheet.insertRule(" [data-anchorjs-icon]::after {   content: attr(data-anchorjs-icon); }",e.sheet.cssRules.length),e.sheet.insertRule(' @font-face {   font-family: "anchorjs-icons";   src: url(data:n/a;base64,AAEAAAALAIAAAwAwT1MvMg8yG2cAAAE4AAAAYGNtYXDp3gC3AAABpAAAAExnYXNwAAAAEAAAA9wAAAAIZ2x5ZlQCcfwAAAH4AAABCGhlYWQHFvHyAAAAvAAAADZoaGVhBnACFwAAAPQAAAAkaG10eASAADEAAAGYAAAADGxvY2EACACEAAAB8AAAAAhtYXhwAAYAVwAAARgAAAAgbmFtZQGOH9cAAAMAAAAAunBvc3QAAwAAAAADvAAAACAAAQAAAAEAAHzE2p9fDzz1AAkEAAAAAADRecUWAAAAANQA6R8AAAAAAoACwAAAAAgAAgAAAAAAAAABAAADwP/AAAACgAAA/9MCrQABAAAAAAAAAAAAAAAAAAAAAwABAAAAAwBVAAIAAAAAAAIAAAAAAAAAAAAAAAAAAAAAAAMCQAGQAAUAAAKZAswAAACPApkCzAAAAesAMwEJAAAAAAAAAAAAAAAAAAAAARAAAAAAAAAAAAAAAAAAAAAAQAAg//0DwP/AAEADwABAAAAAAQAAAAAAAAAAAAAAIAAAAAAAAAIAAAACgAAxAAAAAwAAAAMAAAAcAAEAAwAAABwAAwABAAAAHAAEADAAAAAIAAgAAgAAACDpy//9//8AAAAg6cv//f///+EWNwADAAEAAAAAAAAAAAAAAAAACACEAAEAAAAAAAAAAAAAAAAxAAACAAQARAKAAsAAKwBUAAABIiYnJjQ3NzY2MzIWFxYUBwcGIicmNDc3NjQnJiYjIgYHBwYUFxYUBwYGIwciJicmNDc3NjIXFhQHBwYUFxYWMzI2Nzc2NCcmNDc2MhcWFAcHBgYjARQGDAUtLXoWOR8fORYtLTgKGwoKCjgaGg0gEhIgDXoaGgkJBQwHdR85Fi0tOAobCgoKOBoaDSASEiANehoaCQkKGwotLXoWOR8BMwUFLYEuehYXFxYugC44CQkKGwo4GkoaDQ0NDXoaShoKGwoFBe8XFi6ALjgJCQobCjgaShoNDQ0NehpKGgobCgoKLYEuehYXAAAADACWAAEAAAAAAAEACAAAAAEAAAAAAAIAAwAIAAEAAAAAAAMACAAAAAEAAAAAAAQACAAAAAEAAAAAAAUAAQALAAEAAAAAAAYACAAAAAMAAQQJAAEAEAAMAAMAAQQJAAIABgAcAAMAAQQJAAMAEAAMAAMAAQQJAAQAEAAMAAMAAQQJAAUAAgAiAAMAAQQJAAYAEAAMYW5jaG9yanM0MDBAAGEAbgBjAGgAbwByAGoAcwA0ADAAMABAAAAAAwAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAABAAH//wAP) format("truetype"); }',e.sheet.cssRules.length)}}this.options=A||{},this.elements=[],e(this.options),this.isTouchDevice=function(){return!!("ontouchstart"in window||window.DocumentTouch&&document instanceof DocumentTouch)},this.add=function(A){var i,o,s,c,r,a,h,l,u,d,f,g,p=[];if(e(this.options),"touch"===(g=this.options.visible)&&(g=this.isTouchDevice()?"always":"hover"),A||(A="h2, h3, h4, h5, h6"),0===(i=t(A)).length)return this;for(n(),o=document.querySelectorAll("[id]"),s=[].map.call(o,function(A){return A.id}),r=0;r<i.length;r++)if(this.hasAnchorJSLink(i[r]))p.push(r);else{if(i[r].hasAttribute("id"))c=i[r].getAttribute("id");else if(i[r].hasAttribute("data-anchor-id"))c=i[r].getAttribute("data-anchor-id");else{u=l=this.urlify(i[r].textContent),h=0;do{void 0!==a&&(u=l+"-"+h),a=s.indexOf(u),h+=1}while(-1!==a);a=void 0,s.push(u),i[r].setAttribute("id",u),c=u}d=c.replace(/-/g," "),(f=document.createElement("a")).className="anchorjs-link "+this.options.class,f.href="#"+c,f.setAttribute("aria-label","Anchor link for: "+d),f.setAttribute("data-anchorjs-icon",this.options.icon),"always"===g&&(f.style.opacity="1"),""===this.options.icon&&(f.style.font="1em/1 anchorjs-icons","left"===this.options.placement&&(f.style.lineHeight="inherit")),"left"===this.options.placement?(f.style.position="absolute",f.style.marginLeft="-1em",f.style.paddingRight="0.5em",i[r].insertBefore(f,i[r].firstChild)):(f.style.paddingLeft="0.375em",i[r].appendChild(f))}for(r=0;r<p.length;r++)i.splice(p[r]-r,1);return this.elements=this.elements.concat(i),this},this.remove=function(A){for(var e,n,i=t(A),o=0;o<i.length;o++)(n=i[o].querySelector(".anchorjs-link"))&&(-1!==(e=this.elements.indexOf(i[o]))&&this.elements.splice(e,1),i[o].removeChild(n));return this},this.removeAll=function(){this.remove(this.elements)},this.urlify=function(A){var t=/[& +$,:;=?@"#{}|^~[`%!'<>\]\.\/\(\)\*\\]/g;return this.options.truncate||e(this.options),A.trim().replace(/\'/gi,"").replace(t,"-").replace(/-{2,}/g,"-").substring(0,this.options.truncate).replace(/^-+|-+$/gm,"").toLowerCase()},this.hasAnchorJSLink=function(A){var e=A.firstChild&&(" "+A.firstChild.className+" ").indexOf(" anchorjs-link ")>-1,t=A.lastChild&&(" "+A.lastChild.className+" ").indexOf(" anchorjs-link ")>-1;return e||t||!1}}return A});

// Enable links for selected headers
var anchors = new AnchorJS();
anchors.add("h2, h3, h4")
</script>
<!-- Enable Hypothesis annotation. https://web.hypothes.is/ -->
<script src="https://hypothes.is/embed.js" async></script>
</body>
</html>
