
## Proposal Body (2000 words)

* Base enabling technologies:
  * Low dimensional representations, ProjectR (Elana, Loyal)
  * New methods published by other groups (scVI, etc -  others)
  * Fast & improved quantification  (Rob)

### Aim 1

* Fast search: (in low dimensions, quantifying differences between case and reference maps, twist: shared latent spaces / k-mers)
  * Stephanie: differences b/w maps
  * Rob: new models for UMI deduplication accounting for transcript-level information
    * Parsimony & likelihood based, integrated with gene-level uncertainty
  * Rob: everything FAST! API for search against HCA reference?
  * Casey & Rob: k-mer / quantified latent spaces

### Aim 2

* Transformation of latent spaces
  * Loyal & Elana: search tool for perturbations / signatures in latent-space(s)
  * Elana: latent space transformations for progression? Consider jawns for semi-supervised learning?
  * Loyal transfer of signatures _between_ tissues

### Aim 3

* Reference Cell types
  * Cell-type summarized expression profiles
  * Versioning & provenance of cell types as the reference dataset changes

* Delivery
  * Training / teaching (scRNAseq, low-dimensional representations, RFA-developed tools)
  * Software hardening/testing
  * Bioconductor integration
