
## Proposal Body (2000 words)

* Base enabling technologies:
  * Low dimensional representations (Elana)

Latent space determination relies on computational techniques that learn low dimensional structure from high dimensional data. A wide variety of computational methods for dimensionality reduction have been applied to single cell data. These techniques fall largely into two classes: (1) linear techniques based upon matrix factorization and (2) non-linear techniques based upon manifold learning. Each specific technique selected for analysis will reveal distinct structures in the data. Furthermore, a single technique may learn distinct estimates based upon the dimensionality selected for analysis. Much of the current research emphasizes demonstration of optimality of a single method or dimensionality. In contrast to this prevaling view, we posit that each method or dimensionality reveals a unique set of features associated with a set of *a priori* unknown biological processes. Together, the distribution of the ensemble of low dimensional representations from these methods will uncover the multiple, concurrent biological processes from high dimensional datasets. Previously, we have developed a common language for interpretation of matrix factorization methods to facilitate a unification of latent space methods [@doi:10.1016/j.tig.2018.07.003]. The latent space team in the previous round of funding (containing PIs Fertig, Goff, Greene, and Patro) is also developing standardized output formats for these distinct methods. Together, this previous work will facilitate practical unification, dimensionality estimation, and biological assessment of latent spaces in this seed network. 

ELANA -- LINK IN COGAPS AND WHY WE ARE SELECTING A SINGLE TOOL TO ENABLE A LOT OF STUFF HERE. NOTE GENERALITY TO OTHER METHODS PUBLISHED BY OTHER GROUPS.

  * New methods published by other groups (scVI, etc -  other groups)
  * Fast & improved quantification  (Rob)
  * Incorporation of uncertainty estimates in low dimensional representations (Rob / Mike / Elana) - note this can be done w/o modification in scCoGAPS

### Aim 1

* Fast search: (in low dimensions, quantifying differences between case and reference maps, twist: shared latent spaces / k-mers)
  * Differences b/w maps (Stephanie)
  * New models for UMI deduplication accounting for transcript-level information (Rob)
    * Parsimony & likelihood based, integrated with gene-level uncertainty
  * Everything FAST! API for search against HCA reference? (Rob)
  * k-mer / quantified latent spaces (Casey / Rob)

### Aim 2

* Eschewing marker genes: Practical exploration of the HCA in latent spaces
  * Search tool for perturbations / signatures in latent-space(s) (Loyal)
    - Differential analysis of latent space usage across contexts
  * Latent space transformations for progression? Consider jawns for semi-supervised learning? (Elana)
  * Transfer learning of signatures _between/across_ tissues (Loyal)

* Reference Cell types
  * Cell-type summarized expression profiles (Mike, Loyal)
    - A 'reference catalog' of reduced dimensional representations
  * Versioning & provenance of cell types / features as the reference dataset changes (Mike)
  * 'Power-user' application of latent-space discovery in novel dataset and projection of HCA into new learned spaces.

### Aim 3

* Delivery
  * Training / teaching (scRNAseq, low-dimensional representations, RFA-developed tools) (Tom)
  * Software hardening/testing (Casey - software eng)
  * Bioconductor integration (Stephanie, Mike)
